agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -1484,6 +1484,52 @@ bool monitor_thread::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t event
 
         // If event == Channel_Scan_New_Results_Ready do nothing since is_dump's default is 0
         if (event == Event::Channel_Scan_Dump_Result) {
+            auto msg = static_cast<bwl::sCHANNEL_SCAN_RESULTS_NOTIFICATION *>(data);
+
+            auto in_result  = &(msg->channel_scan_results);
+            auto out_result = &(notification->scan_results());
+
+            // Strings, simple arrays & vectors:
+            string_utils::copy_string(out_result->ssid, in_result->ssid,
+                                      beerocks::message::WIFI_SSID_MAX_LENGTH);
+            std::copy_n(in_result->bssid.oct, sizeof(out_result->bssid.oct), out_result->bssid.oct);
+            std::copy(in_result->basic_data_transfer_rates_kbps.begin(),
+                      in_result->basic_data_transfer_rates_kbps.end(),
+                      out_result->basic_data_transfer_rates_kbps);
+            std::copy(in_result->supported_data_transfer_rates_kbps.begin(),
+                      in_result->supported_data_transfer_rates_kbps.end(),
+                      out_result->supported_data_transfer_rates_kbps);
+
+            // Simple values
+            out_result->channel             = in_result->channel;
+            out_result->signal_strength_dBm = in_result->signal_strength_dBm;
+            out_result->beacon_period_ms    = in_result->beacon_period_ms;
+            out_result->noise_dBm           = in_result->noise_dBm;
+            out_result->dtim_period         = in_result->dtim_period;
+            out_result->channel_utilization = in_result->channel_utilization;
+
+            // Enums
```

Comment:
OK now the previous comment is clear. All these comments are not necessary, they don't add information.

Reply:
I wanted to explain my reason for dividing them into segments and not doing them in-order

Reply:
OK